### ISSUE TITLE
As a channel admin, I would like to view and edit a collection that I have permission to upload to

### ISSUE THREAD

**Issue by pdpinch** [mitodl/odl-video-service#52]
The channel admin should be able to view the list of videos in any channel they have admin permissions for, and upload additional videos to it. 

~~Depends on mitodl/mit-moira#1~~

Depends on #142

**Comment by mbertrand** [comment]
- We need to differentiate between view-only moira lists and admin moira lists.  Currently the Collection model has a single `moira_lists` attribute.  We could add an `admin_moira_lists` attribute to the Collection model, or create a new 'CollectionMoiraList' model that would have many-to-many foreign keys to both MoiraList and Collection plus a permissions attribute ('View', 'Admin').  

- When checking moira permissions, the app will have to retrieve the members of each moira list associated with the collection, and check if the user's email is included.  This could impact performance, maybe the results could be temporarily cached for a small amount of time.

**Comment by pdpinch** [comment]
If we haven't already created a many-to-many for admin moira lists, I would be fine with it being one to one. Particularly for admin permissions, I'm concerned about confusing users with too many lists. 

Caching is worth considering. I think @singingwolfboy told me the moira look-ups could be very slow. However, it should be a separate issue.

**Comment by pdpinch** [comment]
@giocalitri @mbertrand which APIs will need to be updated with this permission?

**Comment by mbertrand** [comment]
`ui.views.CollectionViewSet`

**Comment by mbertrand** [comment]
Who can create collections and how do we specify that?

**Comment by mbertrand** [comment]
- Modify Collection model - `moira_lists` renamed to `view_lists` and create `admin_lists`
- Need permission classes for viewers only and for admins
- Modify the querysets that filter collections by owner only to include moira admin permissions too
- How to optimize querying across collections and moira lists?
- Still allow superuser to do/see everything
- Much of this also applies to #53

**Comment by giocalitri** [comment]
> How to optimize querying across collections and moira lists?

I believe that if we do not address this, the application will become pretty slow if not unusable:
we are going to query the moira list rest API really often (multiple times per request to the OVS website) so we need to find a way to at least cache the results of the call to the moira list rest API in the very short term (like 30 seconds).

**Comment by mbertrand** [comment]
Should 'As a user I should be able to see a list of all collections I have admin access for ' be considered part of this issue or should it be a separate issue?

**Comment by pdpinch** [comment]
We can consider that a separate issue, #152. I'm not convinced it's part of MVP. 

> Should 'As a user I should be able to see a list of all collections I have admin access for ' be considered part of this issue or should it be a separate issue?
